src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -932,13 +932,18 @@ fastrtps::TopicAttributes DataWriterImpl::get_topic_attributes(
 }
 
 OfferedIncompatibleQosStatus& DataWriterImpl::update_offered_incompatible_qos(
-        QosPolicyId_t policy_id)
+        PolicyMask incompatible_policies)
 {
     ++offered_incompatible_qos_status_.total_count;
     ++offered_incompatible_qos_status_.total_count_change;
-    ++offered_incompatible_qos_status_.policies[policy_id].count;
-    offered_incompatible_qos_status_.last_policy_id = policy_id;
-
+    for (fastrtps::rtps::octet id = 0; id < NEXT_QOS_POLICY_ID; ++id)
+    {
+        if (incompatible_policies.test(id))
+        {
+            ++offered_incompatible_qos_status_.policies[static_cast<QosPolicyId_t>(id)].count;
+            offered_incompatible_qos_status_.last_policy_id = static_cast<QosPolicyId_t>(id);
+        }
+    }
```

Comment:
I think we should:
* Start the loop on the first policy, not on the invalid one
* Increase the total counts every time we increase the individual count

Reply:
If I'm understanding the standard correctly, the total counts refer to the number of peers with an incompatible QoS, not to the number of incompatible QoS found in total:
```
Total cumulative number of times the concerned DataWriter
discovered a DataReader for the same Topic with a requested QoS that
is incompatible with that offered by the DataWriter.
```
i.e., if we only have 1 peer, the total count should not be greater than one.

Reply:
You're right.

Starting the loop on the first policy still applies, though.